Deserialisation of Int to BigDecimal fails

Attempting to deserialise an int as a BigDecimal fails with the following exception:

```
Caused by: com.arangodb.velocypack.exception.VPackValueTypeException: Expecting type STRING or DOUBLE
	at com.arangodb.velocypack.VPackSlice.getAsBigDecimal(VPackSlice.java:293) ~[velocypack-2.3.1.jar:?]
	at com.arangodb.jackson.dataformat.velocypack.internal.VPackParser.getDecimalValue(VPackParser.java:289) ~[jackson-dataformat-velocypack-0.1.5.jar:?]
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$BigDecimalDeserializer.deserialize(NumberDeserializers.java:948) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$BigDecimalDeserializer.deserialize(NumberDeserializers.java:927) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:542) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:535) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:419) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1310) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:331) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:164) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4482) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3504) ~[jackson-databind-2.11.0.jar:2.11.0]
	at com.arangodb.VelocyJack.deserialize(VelocyJack.java:112) ~[jackson-dataformat-velocypack-0.1.5.jar:?]
	... 18 more
```

This is caused by the `getAsBigDecimal` method in `VPackSlice.java` not parsing ints.

```
public BigDecimal getAsBigDecimal() {
		if (isString()) {
			return new BigDecimal(getAsString());
		} else if (isDouble()) {
			return BigDecimal.valueOf(getAsDouble());
		} else {
			throw new VPackValueTypeException(ValueType.STRING, ValueType.DOUBLE);
		}
	}
```

--- a/src/main/java/com/arangodb/velocypack/VPackSlice.java
+++ b/src/main/java/com/arangodb/velocypack/VPackSlice.java
@@ -289,6 +289,10 @@ public class VPackSlice implements Serializable {
 			return new BigDecimal(getAsString());
 		} else if (isDouble()) {
 			return BigDecimal.valueOf(getAsDouble());
+		} else if (isSmallInt() || isInt()) {
+			return BigDecimal.valueOf(getAsLong());
+		} else if (isUInt()) {
+			return new BigDecimal(NumberUtil.toBigInteger(vpack, start + 1, length()));
 		} else {
 			throw new VPackValueTypeException(ValueType.STRING, ValueType.DOUBLE);
 		}

--- a/src/test/java/com/arangodb/velocypack/VPackSerializeDeserializeTest.java
+++ b/src/test/java/com/arangodb/velocypack/VPackSerializeDeserializeTest.java
@@ -717,6 +717,7 @@ public class VPackSerializeDeserializeTest {
 	protected static class TestEntityBigNumber {
 		private BigInteger bi = BigInteger.valueOf(1L);
 		private BigDecimal bd = BigDecimal.valueOf(1.5);
+		private BigDecimal bdi = BigDecimal.valueOf(1);
 
 		public BigInteger getBi() {
 			return bi;
@@ -732,6 +733,14 @@ public class VPackSerializeDeserializeTest {
 
 		public void setBd(final BigDecimal bd) {
 			this.bd = bd;
+		}
+
+		public BigDecimal getBdi() {
+			return bdi;
+		}
+
+		public void setBdi(final BigDecimal bdi) {
+			this.bdi = bdi;
 		}
 	}
 
@@ -750,6 +759,11 @@ public class VPackSerializeDeserializeTest {
 			assertThat(bd.isString(), is(true));
 			assertThat(bd.getAsBigDecimal(), is(BigDecimal.valueOf(1.5)));
 		}
+		{
+			final VPackSlice bdi = vpack.get("bdi");
+			assertThat(bdi.isString(), is(true));
+			assertThat(bdi.getAsBigDecimal(), is(BigDecimal.valueOf(1)));
+		}
 	}
 
 	@Test
@@ -759,6 +773,7 @@ public class VPackSerializeDeserializeTest {
 			builder.add(ValueType.OBJECT);
 			builder.add("bi", BigInteger.valueOf(2));
 			builder.add("bd", BigDecimal.valueOf(3.75));
+			builder.add("bdi", BigDecimal.valueOf(3));
 			builder.close();
 		}
 		final VPackSlice vpack = builder.slice();
@@ -766,6 +781,7 @@ public class VPackSerializeDeserializeTest {
 		assertThat(entity, is(notNullValue()));
 		assertThat(entity.bi, is(BigInteger.valueOf(2)));
 		assertThat(entity.bd, is(BigDecimal.valueOf(3.75)));
+		assertThat(entity.bdi, is(BigDecimal.valueOf(3)));
 	}
 
 	@Test
@@ -775,6 +791,20 @@ public class VPackSerializeDeserializeTest {
 		assertThat(fromDouble, is(fromString));
 		assertThat(new VPackBuilder().add(fromDouble).slice().getAsBigDecimal(), is(fromDouble));
 		assertThat(new VPackBuilder().add(fromString).slice().getAsBigDecimal(), is(fromDouble));
+	}
+
+	@Test
+	public void deserializeIntAsBigDecimal() {
+		final int integer = Integer.MAX_VALUE;
+		final BigDecimal fromInt = BigDecimal.valueOf(integer);
+		assertThat(new VPackBuilder().add(integer).slice().getAsBigDecimal(), is(fromInt));
+	}
+
+	@Test
+	public void deserializeLongAsBigDecimal() {
+		final long lng = Long.MAX_VALUE;
+		final BigDecimal fromLng = BigDecimal.valueOf(lng);
+		assertThat(new VPackBuilder().add(lng).slice().getAsBigDecimal(), is(fromLng));
 	}
 
 	protected static class TestEntityArray {

